### PR TITLE
added support for container.image

### PR DESCRIPTION
**Pull request by jinfwhuang**
Added support to do image replacement for a custom resource similar to
```
apiVersion: argoproj.io/v1alpha1
kind: CronWorkflow
metadata:
  name: cwfresource
spec:
  workflowSpec:
    templates:
      - name: stage1
        container:
          image: helloworld-image
      - name: stage2
        container:
          image: helloworld-image

```

Otherwise, see this ugly work-around: (removed private url)

### CODE REVIEW

File: resolver/pkg/resolver.go [hunk 1]
```diff
@@ -79,8 +79,19 @@ type imageTagTransformer struct {
  then loops though all images inside containers session, finds matched ones and update the tag name
 */
 func (pt *imageTagTransformer) findAndReplaceTag(obj map[string]interface{}) error {
-	paths := []string{"containers", "initContainers"}
 	found := false
+
+	// Update contain.image
```
Comment (kzadorozhny) [inline resolver/pkg/resolver.go:84]: `container.image`

File: resolver/pkg/resolver.go [hunk 2]
```diff
@@ -79,8 +79,19 @@ type imageTagTransformer struct {
...
+	_, found = obj["container"]
```
Comment (kzadorozhny) [inline resolver/pkg/resolver.go:85]: Please combine it with `for _, path := range paths` loop. I don't see the reason why this code is separate.

File: examples/helloworld/cwf.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+apiVersion: argoproj.io/v1alpha1
```
Comment (kzadorozhny) [inline examples/helloworld/cwf.yaml:1]: There is E2E test that deploys helloworld resources. Without Argo CRD installed it will fail. Please remove.

Comment (jinfwhuang) [reply examples/helloworld/cwf.yaml:1]: sure. I will remove them.

File: resolver/pkg/resolver.go (hunk 1 above)
Comment (jinfwhuang) [reply resolver/pkg/resolver.go:84]: updated.

File: resolver/pkg/resolver.go (hunk 2 above)
Comment (jinfwhuang) [reply resolver/pkg/resolver.go:85]: because one path is a list and other path is a single map.

**Pull request merged**